asm_riscv/src/lib.rs

```diff
@@ -520,97 +520,96 @@ impl From<I> for u32 {
     }
 }
 
-impl From<u32> for I {
+impl TryFrom<u32> for I {
+    type Error = ();
     // Using match makes it easier to extend code in the future.
     #[allow(clippy::match_single_binding)]
-    fn from(with: u32) -> Self {
+    fn try_from(with: u32) -> Result<Self, Self::Error> {
         match with & 0b1111111 {
             // Load From RAM
             0b0000011 => match I::from_i(with) {
-                (d, 0b000, s, im) => LB { d, s, im },
-                (d, 0b001, s, im) => LH { d, s, im },
-                (d, 0b010, s, im) => LW { d, s, im },
-                (d, 0b100, s, im) => LBU { d, s, im },
-                (d, 0b101, s, im) => LHU { d, s, im },
-                (_, funct, _, _mm) => panic!("Unknown funct3: {}", funct),
```

Comment:
Thanks for opening a PR!  I would like the error to be a `String`, converting the `panic!(...)` statements to `Err(format!(...))` instead of `Err(())` as to not lose the data for debugging.

Reply:
What do you think of something like

```
pub enum ConversionErrorType{
    UnknownFunct3(u32),
    UnknownFunct3Funct7(u32, u32),
    UnknownEnvCtrlTransfer,
    UnknownOpcode(u32),
}

impl TryFrom<u32> for I {
    type Error = ConversionErrorType;
    ...
```

making the error type more parseable than a straight String?

Reply:
I've gone ahead and pushed an example. I'll revert the other change which i guess was caused by rustfmt.